llvm/test/CodeGen/AIE/aie2/GlobalISel/legalize-fcmp-bf16.mir

```diff
@@ -0,0 +1,302 @@
+# NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
+#
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
+# RUN: llc -mtriple aie2 -run-pass=legalizer %s -verify-machineinstrs -o - | FileCheck %s
+
+---
+name:            fcmp_ogt
+body:             |
+  bb.1.entry:
+    liveins: $r1, $r2
+    ; CHECK-LABEL: name: fcmp_ogt
+    ; CHECK: liveins: $r1, $r2
+    ; CHECK-NEXT: {{  $}}
+    ; CHECK-NEXT: [[COPY:%[0-9]+]]:_(s32) = COPY $r1
+    ; CHECK-NEXT: [[COPY1:%[0-9]+]]:_(s32) = COPY $r2
+    ; CHECK-NEXT: [[DEF:%[0-9]+]]:_(<32 x s16>) = G_IMPLICIT_DEF
+    ; CHECK-NEXT: [[C:%[0-9]+]]:_(s32) = G_CONSTANT i32 0
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[AIE_INSERT_VECTOR_ELT1:%[0-9]+]]:_(<32 x s16>) = G_AIE_INSERT_VECTOR_ELT [[DEF]], [[COPY1]](s32), [[C]](s32)
+    ; CHECK-NEXT: [[INT:%[0-9]+]]:_(s32) = G_INTRINSIC intrinsic(@llvm.aie2.vltbf16), [[AIE_INSERT_VECTOR_ELT1]](<32 x s16>), [[AIE_INSERT_VECTOR_ELT]](<32 x s16>)
+    ; CHECK-NEXT: [[C1:%[0-9]+]]:_(s32) = G_CONSTANT i32 1
+    ; CHECK-NEXT: [[AND:%[0-9]+]]:_(s32) = G_AND [[INT]], [[C1]]
+    ; CHECK-NEXT: $r0 = COPY [[AND]](s32)
+    ; CHECK-NEXT: PseudoRET implicit $lr, implicit $r0
+    %3:_(s32) = COPY $r1
+    %1:_(s16) = G_TRUNC %3:_(s32)
```

Comment:
Do we need the G_TRUNC?

Reply:
`half` might add to the confusion here, since we are handling `bfloat16`. Can we use `G_CONSTANT bfloat`?

If not, we should keep the G_TRUNC. bfloat is passed in a 32-bit register, and then truncated in the callee

Reply:
I didn't change..